### PR TITLE
Feature/kyv 233 right of purchase

### PR DESCRIPTION
**Pull request by severikupari1**

### CODE REVIEW

File: common/src/main/java/fi/hel/verkkokauppa/common/rest/RestServiceClient.java [hunk 1]
```diff
@@ -29,13 +29,21 @@
     public JSONObject makeGetCall(String url) {
         WebClient client = getClient();
         JSONObject response = queryJsonService(client, url);
-        return response;
+        if (response == null) {
```
Comment (severikupari1) [inline common/src/main/java/fi/hel/verkkokauppa/common/rest/RestServiceClient.java:32]: These fixed NullPointerException when JsonObject is tried to make from null value.

File: mockproductmanagement/src/main/java/fi/hel/verkkokauppa/mockproductmanagement/api/MockOrderRightOfPurchaseController.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("mock/**/order/")
```
Comment (severikupari1) [inline mockproductmanagement/src/main/java/fi/hel/verkkokauppa/mockproductmanagement/api/MockOrderRightOfPurchaseController.java:15]: This is gonna be deleted in another pull request, KYV-393-

File: mockproductmanagement/pom.xml [hunk 3]
```diff
@@ -57,7 +57,13 @@
 			<artifactId>spring-integration-test</artifactId>
 			<scope>test</scope>
 		</dependency>
-	</dependencies>
+        <dependency>
+            <groupId>fi.hel</groupId>
+            <artifactId>common</artifactId>
+            <version>0.0.2-SNAPSHOT</version>
+            <scope>compile</scope>
```
Comment (hleisma) [inline mockproductmanagement/pom.xml:64]: This is the default scope, used if none is specified.

File: orderapi/src/main/java/fi/hel/verkkokauppa/order/service/rightOfPurchase/OrderRightOfPurchaseService.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                namespace,
+                ServiceConfigurationKeys.ORDER_RIGHT_OF_PURCHASE_URL
+        );
+    }
+
+    public ResponseEntity<Boolean> canPurchase(OrderAggregateDto order) throws JsonProcessingException {
+        // The Backend returns TRUE
+        // by default if the ORDER_RIGHT_OF_PURCHASE_IS_ACTIVE service configuration is not
+        // defined in the service configuration.
+        if (!isActive() || !hasOrderRightOfPurchaseUrl()) {
```
Comment (hleisma) [inline orderapi/src/main/java/fi/hel/verkkokauppa/order/service/rightOfPurchase/OrderRightOfPurchaseService.java:64]: Careful with negations like this. Would be easier to trust if wrapped as: if (noRightOfPurchaseLimitationsDefined()) return true;

**Review by hleisma — APPROVED** [review]

File: orderapi/src/main/java/fi/hel/verkkokauppa/order/service/rightOfPurchase/OrderRightOfPurchaseService.java (hunk 4 above)
Comment (severikupari1) [reply orderapi/src/main/java/fi/hel/verkkokauppa/order/service/rightOfPurchase/OrderRightOfPurchaseService.java:64]: Thats correct, I should make new function to wrap this condition.
